From 13c20625b6e04316ed685064969ea2f7acc0a37e Mon Sep 17 00:00:00 2001
From: Vidya <vvelumuri@marvell.com>
Date: Mon, 21 Dec 2020 08:58:30 +0000
Subject: [PATCH 1052/1921] octeontx2-af: configure npc for cn10k to allow
 packets from cpt

The higher bits in the channel number represents the CPT
channel number. Mask out these higher bits in the npc configuration
to allow packets from cpt for parsing.

Signed-off-by: Vidya <vvelumuri@marvell.com>
Change-Id: Ibe6fd67b5af603f3b3252c664a9d3bbf0f6b0bc4
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/42605
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu_npc.c | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_npc.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_npc.c
index ebe3f63eb329..abdb517610a5 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_npc.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_npc.c
@@ -695,7 +695,17 @@ void rvu_npc_install_promisc_entry(struct rvu *rvu, u16 pcifunc,
 		req.features = BIT_ULL(NPC_DMAC);
 	}
 
-	req.chan_mask = 0xFFFU;
+	/* For cn10k the upper two bits bits of the channel number are
+	 * cpt channel number. with masking out these bits in the
+	 * mcam entry, same entry used for NIX will allow packets
+	 * received from cpt for parsing.
+	 */
+	if (!is_rvu_otx2(rvu)) {
+		req.chan_mask = NIX_CHAN_CPT_X2P_MASK;
+	} else {
+		req.chan_mask = 0xFFFU;
+	}
+
 	if (chan_cnt > 1) {
 		if (!is_power_of_2(chan_cnt)) {
 			dev_err(rvu->dev,
-- 
2.31.1

